src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/DecryptTests.cs

```diff
@@ -583,14 +538,91 @@ private static void VerifySimpleDecrypt(byte[] encodedMessage, CertLoader certLo
                 if (cert == null)
                     return; // Sorry - CertLoader is not configured to load certs with private keys - we've tested as much as we can.
 
-                X509Certificate2Collection extraStore = new X509Certificate2Collection(cert);
-                ecms.Decrypt(extraStore);
+#if netcoreapp // API not present on netfx
+                if (_useExplicitPrivateKey)
+                {
+                    using (X509Certificate2 pubCert = certLoader.GetCertificate())
+                    {
+                        RecipientInfo recipient = ecms.RecipientInfos.Cast<RecipientInfo>().Where((r) => r.RecipientIdentifier.MatchesCertificate(cert)).Single();
+                        ecms.Decrypt(recipient, cert.PrivateKey);
+                    }
+                }
+                else
+#endif
+                {
+                    X509Certificate2Collection extraStore = new X509Certificate2Collection(cert);
+                    ecms.Decrypt(extraStore);
+                }
                 ContentInfo contentInfo = ecms.ContentInfo;
                 Assert.Equal(expectedContent.ContentType.Value, contentInfo.ContentType.Value);
                 Assert.Equal<byte>(expectedContent.Content, contentInfo.Content);
             }
         }
-    }
-}
 
+        public class DecryptTestsUsingCertWithPrivateKey : DecryptTests
+        {
+            public DecryptTestsUsingCertWithPrivateKey() : base(false) { }
+
+            [ConditionalFact(nameof(SupportsCngCertificates))]
+            [OuterLoop(/* Leaks key on disk if interrupted */)]
+            public static void DecryptMultipleRecipients()
+            {
+                // Force Decrypt() to try multiple recipients. Ensure that a failure to find a matching cert in one doesn't cause it to quit early.
+
+                CertLoader[] certLoaders = new CertLoader[]
+                {
+                Certificates.RSAKeyTransfer1,
+                Certificates.RSAKeyTransfer2,
+                Certificates.RSAKeyTransfer3,
```

Comment:
Looks like indentation got messed up during the move.

Reply:
interestingly VS doesn't autofix it as well...